content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,205 @@
+
+=== How to run the Plugin
+See https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+and set credentials, owner and repository on your own and you will be good to go.
+
+=== What is avalilable now
+1. User will be able to select to use Jenkinsfile.yaml for build.
+2. Git Push step
+3. harvest results and reports [and post in the PR]
+ a. junit()
+ b. findbugs()
+ c. archiveArtifacts()
+4. Basic interface to parse and get build specifications from YAML file.
+
+For now the plugin is bulding branches and Pull request both using Jnekinsfile.yaml.
+This will be fixed in next coding phase.
+
+=== Things decided
+1. To build the plugin on the top of multibranch pipeline plugin. As that plugin has implementation of
+  a. Nice interface to show diffrent branch and pull requests build seperately with use of suitable plugins like Github, Bitbucket.
+  b. Detect trusted revisions in a repository.
+  c. Pblishing of build status to the repository.
+
+2. Convert the YAML configuration to declarative pipeline.
+3. User will provide path to the script relative to the root directory of the repository
+  without extension (.sh or .bat) in the YAML file. The plugin will generate pipeline script to detect the
+  platform and call .sh or .bat script.
+
+  Example:
+    Path provided: ./scripts/hello
+    a. On UNIX machine “./scripts/hello.sh” will be called
+    b. On non-UNIX machine “./scripts/hello.bat” will be called.
+
+=== Implementations till now
+1. Git Push step: To push the changes of PR to the target branch. This is implemented
+using git-plugin, PushCommand is used for this from git-plugin.
+2. Basic conversion of YAML to Declarative Pipeline: A class YamlToPipeline class
+is written which will load the "Jenkinsfile.yaml" and make use of PipelineSnippetGenerator class
+to generate Declarative pipeline code.
+3. Reporting of results.
+4. Plugin is using Yaml from target branch rigth now. (May be this needs some discussion, example: what if PR contains changes in Jenkinsfile.yaml)
+=== Jenkinsfile.yaml exapmle
+Write now no format has been decided for the yaml file. But for thes demo I am using below yaml Jenkinsfile
+[source,yaml]
+artifactPublishingConfig:
+    host: 192.32.52.12
+    user: user53
+    credentialId: dummyGitRepo
+agent:
+    dockerImage: maven:3.5.3-jdk-8
+    args: -v /tmp:/tmp
+testResultPaths:
+    - target/surefire-reports/*.xml
+findBugs: target/*.xml
+stages:
+    - name: First
+      scripts:
+        -   ./scripts/hello
+    - name: Build
+      scripts:
+        -   ./scripts/build
+    - name: Tests
+      scripts:
+        -   ./scripts/test
+archiveArtifacts:
+    - Jenkinsfile.yaml
+    - scripts/hello.sh
+publishArtifacts:
+    - from: Jenkinsfile.yaml
+      to: ~/archives
+    - from: scripts/hello.sh
+      to: ~/archives
+
+Below pipeline code will be generated using above yaml file.
+
+[source, groovy]
```

Comment:
I am not able to fit pipeline code in the screen so that can be read comfortably. I will need to split the pipeline code and upload two pictures. Will that be ok ?

Reply:
My proposal was to provide example of a successful Pipeline run. It could be taken from Blue Ocean which shows fancy layouts.

The code itself... o need to put it to screenshots. Sorry for confusion

Reply:
I am able to do this.

Reply:
I may be also possible to show screenshots of tests being reported